src/HDF5Msg.chpl

```diff
@@ -24,1759 +26,1622 @@ module HDF5Msg {
+                           errorClass="FileNotFoundError");
+                }
 
-                    /* 
-                    * To prepare for writing the charList to hdf5, do the following, if applicable:
-                    * 1. Remove the characters shuffled left to the previous locale
-                    * 2. Remove the characters shuffled right to the next locale
-                    * 3. If (2) does not apply, add null uint(8) char to end of the charList
-                    */
-                    var shuffleLeftIndex = shuffleLeftIndices[idx]:int;
-                    var shuffleRightIndex = shuffleRightIndices[idx]:int;
+                // validate that the dataset does not already exist
+                validateDataset(file_id, f, dset_name);
 
-                    /*
-                    * Verify if the current locale (idx) contains chars shuffled left to the previous
-                    * locale (idx-1) by checking the shuffleLeftIndex, the number of strings in 
-                    * the current locale, and whether the preceding locale ends with a complete
-                    * string. If (1) the shuffleLeftIndex > -1, (2) this locale contains 2..n
-                    * strings, and (3) the previous locale does not end with a complete string,
-                    * this means the charList contains chars that were shuffled left to complete
-                    * the last string in the previous locale (idx-1). If so, generate
-                    * a new charList that has those values sliced out. 
-                    */
-                    if shuffleLeftIndex > -1 && !isSingleString[idx]
-                                                        && !endsWithCompleteString[idx-1] {
-                        /*
-                        * Since the leading slice was used to complete the last string in
-                        * the previous locale (idx-1), slice those chars from the charList
-                        */
-                        charList = new list(adjustForLeftShuffle(shuffleLeftIndex,charList));    
+                var shape_sym: borrowed GenSymEntry = getGenericTypedArrayEntry(msgArgs.getValueOf("shape"), st);
+                var shape = toSymEntry(shape_sym, int);
+                var dims: int = * reduce shape.a;
 
-                        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                'adjusted locale %i for left shuffle to %i'.format(idx,idx-1)); 
-                    } else {
-                        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                'no left shuffle adjustment for locale %i'.format(idx));
+                var dtype: C_HDF5.hid_t;
+                
+                select entryDtype {
+                    when DType.Int64 {
+                        var flat = toSymEntry(toGenSymEntry(entry), int);
+                        var localFlat = new lowLevelLocalizingSlice(flat.a, 0..#flat.size);
```

Comment:
Not suggesting we make any changes now, but I think with very large arrays that can't all be loaded into memory on a single node, we may want to just serially loop over the locales and then append to a single file so that no nodes ever have to hold the entire array.

Also, not sure what I would expect to perform better given the extra communication needed here. I think it would just depend on what the intended use case is of this function (e.g., are we writing only relatively small pdarrays to single files?)

Reply:
@bmcdonald3 - I believe the intended case here is for small arrays. However, I will verify that and if future modifications are required for this to perform more optimally, I will add issues to handle those changes.